src/binding/petsc4py/src/petsc4py/PETSc/Mat.pyx

```diff
@@ -1481,7 +2162,42 @@ cdef class Mat(Object):
 
     # matrix-matrix product
 
-    def matMult(self, Mat mat, Mat result=None, fill=None):
+    def matMult(
+        self,
+        Mat mat,
+        Mat result=None,
+        fill: float | None = None
+    ) -> Mat:
+        """Performs matrix-matrix multiplication C=AB.
+
+        Neighbor-wise collective.
```

Comment:
I think we agreed on Collective, Not collective and (possibly) Logically collective.

Reply:
Neighborwise collective was added after you left, but we agreed on the American version as the other docs are written this way. I will fix.